game.py

```diff
@@ -17,25 +17,56 @@ def snowman(snowman_word):
     """Complete the snowman function
     replace "pass" below with your own code
     It should print 'Congratulations, you win!'
-    If the player wins and, 
+    If the player wins and,
     'Sorry, you lose! The word was {snowman_word}' if the player loses
     """
-    pass
+    wrong_guesses_list = []
+    wrong_guesses_count = len(wrong_guesses_list)
+    correct_letter_guess_statuses = build_letter_status_dict(snowman_word)
+    victory = False
+    while wrong_guesses_count < SNOWMAN_MAX_WRONG_GUESSES and not victory:
+        user_guess = get_letter_from_user(correct_letter_guess_statuses, wrong_guesses_list)
+        if user_guess not in snowman_word:
+            wrong_guesses_list.append(user_guess)
+            wrong_guesses_count = len(wrong_guesses_list)
+            print_snowman_graphic(wrong_guesses_count)
+
+        else:
+            correct_letter_guess_statuses[user_guess] = True
+            print_word_progress_string(snowman_word, correct_letter_guess_statuses)
+            victory = is_word_guessed(snowman_word, correct_letter_guess_statuses)
```

Comment:
`victory` is really just a flag containing either `True` or `False`. You use this flag to determine if the `while` loop should terminate or not next iteration. We could alternatively do the following: 
```py
if is_word_guessed: 
    break
```
This would remove the `and not victory` portion of your while loop. However, you need to consider readability of your code as well. Which one more readily communicates to developers that this while loop is `break` condition is partially dependent on if the word is guessed? @Bipentium25